sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -4282,15 +4311,14 @@ pi_result piEnqueueMemBufferMap(pi_queue Queue, pi_mem Buffer,
   ze_fence_handle_t ZeFence = nullptr;
   ze_event_handle_t ZeEvent = nullptr;
 
+  // Lock automatically releases when this goes out of scope.
+  std::lock_guard<std::mutex> lock(Queue->PiQueueMutex);
+
```

Comment:
I can't move this up because piEventsWait is trying to acquire the lock. So reverting this change.

Reply:
you can still have this lock specially for createEventAndAssociateQueue, before line 4318

Reply:
Yes, that's what did. I moved it inside the { } so the lock_guard is only used for createEventAndAssociateQueue. There is a separate lock_guard in the original line below.